ui/src/app/components/popups.tsx

```diff
@@ -43,7 +43,41 @@ const styles = {
   },
 };
```

Comment:
no changes observed in behavior in the popups/tooltips/etc in this file.  Just adding types.